brainscore_vision/model_helpers/activations/core.py

```diff
@@ -265,15 +439,62 @@ def attach_stimulus_set_meta(assembly, stimulus_set):
     stimulus_paths = [str(stimulus_set.get_stimulus(stimulus_id)) for stimulus_id in stimulus_set['stimulus_id']]
     stimulus_paths = [lstrip_local(path) for path in stimulus_paths]
     assembly_paths = [lstrip_local(path) for path in assembly['stimulus_path'].values]
+
     assert (np.array(assembly_paths) == np.array(stimulus_paths)).all()
-    assembly['stimulus_path'] = stimulus_set['stimulus_id'].values
+    assembly = assembly.reset_index('presentation')
+    assembly.assign_coords(stimulus_path=('presentation', stimulus_set['stimulus_id'].values))
+
     assembly = assembly.rename({'stimulus_path': 'stimulus_id'})
+
+    all_columns = []
     for column in stimulus_set.columns:
-        assembly[column] = 'stimulus_id', stimulus_set[column].values
-    assembly = assembly.stack(presentation=('stimulus_id',))
+        assembly = assembly.assign_coords({column: ('presentation', stimulus_set[column].values)})
+        all_columns.append(column)
+    if 'stimulus_id' in all_columns:
+        all_columns.remove('stimulus_id')
+    if 'stimulus_path2' in all_columns:
+        all_columns.remove('stimulus_path2')
+    assembly = assembly.set_index(presentation=['stimulus_id', 'stimulus_path2'] + all_columns)
```

Comment:
I'm confused what this does. I guess this somehow tries to work around xarray multi-indexing? Needs at least a code comment, maybe can be simplified